wrappers/src/results_cs.cpp

```diff
@@ -124,20 +123,21 @@ REALM_EXPORT Results* results_get_filtered_results(const Results& results, uint1
 {
     return handle_errors(ex, [&]() {
         Utf16StringAccessor query_string(query_buf, query_len);
-        auto query = results.get_query();
         auto const &realm = results.get_realm();
 
-        parser::ParserResult result = parser::parse(query_string.to_string());
-
-        parser::KeyPathMapping mapping;
+        query_parser::KeyPathMapping mapping;
         realm::populate_keypath_mapping(mapping, *realm);
 
-        query_builder::NoArguments no_args;
-        query_builder::apply_predicate(query, result.predicate, no_args, mapping);
+        query_parser::NoArguments no_args;
+        auto query = results.get_table()->query(query_string, no_args, mapping);
+        auto ordering = query.get_ordering();
 
-        DescriptorOrdering ordering;
-        query_builder::apply_ordering(ordering, query.get_table(), result.ordering);
-        return new Results(realm, std::move(query), std::move(ordering));
+        if (ordering) {
```

Comment:
If ordering is `null`, you should probably fall back to the existing ordering defined on the results: `DescriptorOrdering ordering = query.get_ordering() ? *query.get_ordering() :results. get_descriptor_ordering();` (not sure if C++ will like this syntax, but I hope the idea is clear).